### PR TITLE
Add Document MQM support

### PR DESCRIPTION
**Pull request by zouharvi**
As per #122 and email discussion, this is rebased to `release`.

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -1,5 +1,5 @@
 beautifulsoup4
-Django<4
```
Comment (snukky) [inline requirements.txt:2]: Do we require anything that Django 5 introduced? If not, I would skip this change for now as it has a cascading effect on requirements in our standard WebApp configuration to Python version (easy) and PostgreSQL (not easy).

File: Campaign/management/commands/ProcessCampaignData.py [hunk 2]
```diff
@@ -62,15 +62,12 @@ def _process_campaign_data(campaign, batch_user, campaign_type, max_count):
...
-
         except Exception as e:
             raise CommandError(e)
-
         finally:
             batch_data.dataReady = True
             batch_data.activate()
             batch_data.save()
 
-    print('Campaign activated')
```
Comment (snukky) [inline Campaign/management/commands/ProcessCampaignData.py:74]: Just checking, was this removed accidentally?

Comment (zouharvi) [reply Campaign/management/commands/ProcessCampaignData.py:74]: Must've happened during the rebase. Latest commit adds this back.

File: requirements.txt (hunk 1 above)
Comment (snukky) [reply requirements.txt:2]: Okay, thanks. I may check Django 4 then because it seems to still support Python 3.8

Comment (zouharvi) [reply requirements.txt:2]: Didn't run any tests but this branch works as-is with django 4.2.

File: EvalData/models/direct_assessment_document.py [hunk 3]
```diff
@@ -496,6 +490,10 @@ class DirectAssessmentDocumentResult(BaseAssessmentResult):
         verbose_name=_('Score'), help_text=_('(value in range=[1,100])')
     )
 
+    mqm = models.TextField(
+        verbose_name=_('MQM'), help_text=_('MQM JSON string')
```
Comment (snukky) [inline EvalData/models/direct_assessment_document.py:494]: It seems that this change requires providing a default value for existing data.

File: EvalData/models/base_models.py [hunk 4]
```diff
@@ -645,6 +646,12 @@ class TextPair(EvalItem):
         verbose_name=_('Target text'),
     )
 
+    # user for AI-assisted annotation
+    mqm = models.TextField(
+        blank=True,
+        verbose_name=_('MQM Annotations'),
```
Comment (snukky) [inline EvalData/models/base_models.py:652]: It seems that this change requires providing a default value for existing data.

File: Scripts/create_wmt22_tasks.py
Comment (snukky) [inline Scripts/create_wmt22_tasks.py:1]: This script is used in `RegressionTests/` but they are not modified. It seems that two other tests are failing due to other reasons. We should make sure they work with the proposed changes.

File: requirements.txt (hunk 1 above)
Comment (snukky) [reply requirements.txt:2]: Confirmed. We still need to take a look at migrations and regression tests.

File: EvalData/models/direct_assessment_document.py (hunk 3 above)
Comment (zouharvi) [reply EvalData/models/direct_assessment_document.py:494]: The default should be `[]`. What's the proper syntax? `default="[]"` as an argument?

File: Appraise/settings.py [hunk 5]
```diff
@@ -117,7 +117,7 @@
         INSTALLED_APPS.append('debug_toolbar')
         warnings.warn('Enabled Django Debug Toolbar in installed apps')
     except ImportError:
-        warnings.warn('Django Debug Toolbar not installed')
+        pass
```
Comment (snukky) [inline Appraise/settings.py:120]: I know this message is too frequent and quite annoying, but I think it may be better to make life quality improvement changes in a separate PR.

File: Campaign/management/commands/ProcessCampaignData.py (hunk 2 above)
Comment (snukky) [reply Campaign/management/commands/ProcessCampaignData.py:74]: Thanks.

File: Dashboard/management/commands/CreateInviteTokens.py [hunk 6]
```diff
@@ -85,9 +85,7 @@ def handle(self, *args, **options):
             new_group.save()
 
         if number_of_tokens < 1 or number_of_tokens > 50:
-            _msg = '{0}Specified number of tokens is insane: {1}'.format(
-                WARNING_MSG, number_of_tokens
-            )
+            _msg = f'{WARNING_MSG}Specified number of tokens is insane: {number_of_tokens}'
```
Comment (snukky) [inline Dashboard/management/commands/CreateInviteTokens.py:88]: I'm not a fan of the line length limit of only 88 lines, but it's always good to have some auto formatting even if not perfect, because it makes so many things easier. Let's remember to run Black, see commands in `Makefile` that I added today.

File: EvalData/admin.py [hunk 7]
```diff
@@ -272,6 +274,7 @@ class TextPairAdmin(BaseMetadataAdmin):
         'sourceText',
         'targetID',
         'targetText',
+        'mqm',
```
Comment (snukky) [inline EvalData/admin.py:277]: I would suggest a more descriptive name. Maybe `mqmAnnotations`?

File: EvalData/models/base_models.py (hunk 4 above)
Comment (snukky) [reply EvalData/models/base_models.py:652]: I propose to fix it as in https://github.com/AppraiseDev/Appraise/tree/romang/develop-mqm

File: EvalData/models/direct_assessment_document.py (hunk 3 above)
Comment (snukky) [reply EvalData/models/direct_assessment_document.py:494]: Yes, I propose a fix here: https://github.com/AppraiseDev/Appraise/tree/romang/develop-mqm

File: EvalView/views.py [hunk 8]
```diff
@@ -38,6 +40,8 @@
 LOGGER = _get_logger(name=__name__)
 
 # pylint: disable=C0103,C0330
+
```
Comment (snukky) [inline EvalView/views.py:43]: I think pylint annotations must be next to the function they apply to? Please check.

File: EvalView/views.py [hunk 9]
```diff
@@ -244,7 +250,8 @@ def direct_assessment(request, code=None, campaign_name=None):
             '(middle) or preference for <em>Candidate B</em> (right).'
         )
 
-    campaign_opts = campaign.campaignOptions or ""
+    campaign_opts = set((campaign.campaignOptions or "").lower().split(";"))
```
Comment (snukky) [inline EvalView/views.py:253]: I think this changes the assumption that we had (undocumented, but in use): `in` operator on a string allowed for some variations, e.g. `namedentit` could be triggered by both `NamedEntityEval;foo;bar` or `NamedEntities`. 

Hm, this may be the reason why few regression tests fail.

**Review by snukky — COMMENTED** [review]
Thanks a lot. I added few comments, but because it's a big PR, I want to review `EvalView/views.py` again more carefully later. For now, I tried to fix some issues in the `romang/develop-mqm` branch. I merged your branch there. Can you take a look and check regression tests locally? 

I also wonder why the automatically generated migrations made so many changes there. Any suggestion?

**Comment by snukky** [comment]
FYI, some of proposed changes are in #128. I think we can work on top of it if they are useful.

**Comment by zouharvi** [comment]
Closing in favor of #128.
